components/sections/Projects/ProjectItem.jsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useRef, useState } from "react";
+import ProjectTitle from "./ProjectTitle";
+
+/* eslint-disable @next/next/no-img-element */
+const ProjectItem = ({ name, imgUri, description, projectUrl, setActive, active, index }) => {
+  const imgRef = useRef();
+  const [width, setWidth] = useState();
+
+  useEffect(() =>{
+    (() =>{
+      window && setWidth(window.innerWidth)
+    })()
```

Comment:
The use of IIFE is not necessary here 
```suggestion
    useEffect(() =>{
      typeOf window !== undefined && setWidth(window.innerWidth)
     }
    ,[])
```